cmd/quadlet/main.go

```diff
@@ -103,28 +103,58 @@ func Debugf(format string, a ...interface{}) {
 func getUnitDirs(rootless bool) []string {
 	// Allow overriding source dir, this is mainly for the CI tests
 	unitDirsEnv := os.Getenv("QUADLET_UNIT_DIRS")
+	dirs := make([]string, 0)
+
 	if len(unitDirsEnv) > 0 {
-		return strings.Split(unitDirsEnv, ":")
+		if !isValidFilePath(unitDirsEnv) {
+			Logf("%s not a valid file path", unitDirsEnv)
+			return nil
+		}
+		for _, eachUnitDir := range strings.Split(unitDirsEnv, ":") {
+			dirs = appendSubPaths(dirs, eachUnitDir)
+		}
+		return dirs
 	}
 
-	dirs := make([]string, 0)
 	if rootless {
 		configDir, err := os.UserConfigDir()
 		if err != nil {
 			fmt.Fprintf(os.Stderr, "Warning: %v", err)
 			return nil
 		}
-		dirs = append(dirs, path.Join(configDir, "containers/systemd"))
+		dirs = appendSubPaths(dirs, path.Join(configDir, "containers/systemd"))
 		u, err := user.Current()
 		if err == nil {
-			dirs = append(dirs, filepath.Join(quadlet.UnitDirAdmin, "users", u.Uid))
+			dirs = appendSubPaths(dirs, filepath.Join(quadlet.UnitDirAdmin, "users", u.Uid))
 		} else {
 			fmt.Fprintf(os.Stderr, "Warning: %v", err)
 		}
 		return append(dirs, filepath.Join(quadlet.UnitDirAdmin, "users"))
 	}
 	dirs = append(dirs, quadlet.UnitDirAdmin)
-	return append(dirs, quadlet.UnitDirDistro)
+	return appendSubPaths(dirs, quadlet.UnitDirDistro)
+}
+
+func isValidFilePath(filePath string) bool {
+	// Ensure the file path is an absolute path and exists
+	_, err := filepath.Abs(filePath)
```

Comment:
`Abs` does not validate that the path exists. It guaranties that you get an absolute path by checking if the path is absolute (i.e. starts with `/`) or relative. In the former case, it returns the value as is and in the latter it creates an absolute path by appending the current running directory to the relative path. In both cases, it does not check anything about the validity of the path.
`filepath` also includes an `IsAbs` function that returns `true` if the path is absolute. But, similarly, it does not check that the path exists.

Reply:
I've unresolved this comment since as I wrote here, `Abs` makes an absolute path from a path that is either relative or already absolute.
In fact, [IsAbs](https://pkg.go.dev/path/filepath#IsAbs) is what you're looking for.
Now, please note that if all what this function does is call `IsAbs` you should just call the latter directly without implementing the formar.